packages/ember-cli-eyeglass/src/broccoli-ln-s.ts

```diff
@@ -0,0 +1,82 @@
+import BroccoliPlugin = require("broccoli-plugin");
+import path = require("path");
+import * as fs from "fs-extra";
+
+type EnsureSymlinkSync = (srcFile: string, destLink: string) => void;
+/* eslint-disable-next-line @typescript-eslint/no-var-requires */
+const ensureSymlink: EnsureSymlinkSync = require("ensure-symlink");
+
+/**
+ * An object where the keys are the files that will be created in the tree and
+ * the values are the source files.
+ * 
+ * @interface FileMap
+ */
+interface FileMap {
+  [relativePath: string]: string;
+}
+
+type BroccoliSymbolicLinkerOptions =
+  Pick<BroccoliPlugin.BroccoliPluginOptions, "annotation" | "persistentOutput">;
+
+/**
+ * Creates symlinks to the source files specified.
+ * 
+ *  BroccoliSymbolicLinker
+ */
+export class BroccoliSymbolicLinker extends BroccoliPlugin {
+  files: FileMap;
+  constructor(fileMap?: FileMap | undefined, options: BroccoliSymbolicLinkerOptions = {}) {
+    let pluginOpts: BroccoliPlugin.BroccoliPluginOptions = {needsCache: false};
+    Object.assign(pluginOpts, options);
+    super([], pluginOpts);
+    this.files = Object.assign({}, fileMap);
+  }
+  reset(fileMap?: FileMap | undefined): void {
+    this.files = Object.assign({}, fileMap);
+  }
+  /**
+   * Record that a symlink should be created from src to dest.
+   *
+   * This can be called many times before the build method is invoked.
+   * Calling it after will not have an effect until the next time build() is
+   * invoked.
+   *
+   * @param src The file that should be symlinked into the tree.
+   * @param dest the relative path from the tree's root to the location of the
+   *   symlink. the filename does not have to be the same.
+   * @returns the absolute path to the location where the symlink will be created.
+   */
+  // eslint-disable-next-line @typescript-eslint/camelcase
+  ln_s(src: string, dest: string): string {
+    this.files[dest] = src;
+    return path.join(this.outputPath, dest);
+  }
+  /**
+   * Returns the number of symlinks that will be created.
+   */
+  numberOfFiles(): number {
+    return Object.keys(this.files).length;
+  }
+  /**
+   * Create the symlinks. Directories to them will be created as necessary.
+   */
+  build(): void {
```

Comment:
As implemented, I believe this will for each rebuild rebuild it's output entirely. This may result downstream plugins into having to do extra work, as they assume their input has changed.

One potentially simple solution, is to skip the work if there are no changes in `this.files` since last time.

Reply:
Further reading suggests yes, there is likely some missing book-keeping here.

Reply:
The tree is reset in postBuild causing `this.files` to become empty.

Reply:
Yes, but combined with `persistentOutput: true`, the contents `this.outputPath`  may not realize deletions.

Reply:
You're correct. It won't delete. I decided to not write that code. As I explained in a different comment, the cost of doing that is a lot of complicated code and bookkeeping and the risk of leaving those files in place as a stale cache is very low in practice because there won't be any dangling references to the asset in the code we produce.